### PR TITLE
Introduce Night Light applet

### PR DESCRIPTION
**Pull request by cybre**

### CODE REVIEW

File: src/applets/night-light/indicator_window.ui [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+              <packing>
+                <property name="left_attach">1</property>
+                <property name="top_attach">0</property>
+              </packing>
+            </child>
+            <child>
+              <object class="GtkLabel">
+                <property name="visible">True</property>
+                <property name="can_focus">False</property>
+                <property name="tooltip_text" translatable="yes">Service to use for uploading screenshots</property>
```
Comment (ikeydoherty) [inline src/applets/night-light/indicator_window.ui]: Eh?

File: src/applets/night-light/indicator_window.ui [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+              <object class="GtkBox">
+                <property name="visible">True</property>
+                <property name="can_focus">False</property>
+                <property name="spacing">10</property>
+                <child>
+                  <object class="GtkLabel">
+                    <property name="visible">True</property>
+                    <property name="can_focus">False</property>
+                    <property name="halign">start</property>
+                    <property name="label" translatable="yes">&lt;span font="11"&gt;Night Light&lt;/span&gt;</property>
```
Comment (ikeydoherty) [inline src/applets/night-light/indicator_window.ui]: This isn't translatable. Not remotely.

File: src/applets/night-light/com.solus-project.night-light.gresource.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<gresources>
+        <gresource prefix="/com/solus-project/night-light">
```
Comment (ikeydoherty) [inline src/applets/night-light/com.solus-project.night-light.gresource.xml]: For new applets please use `/org/budgie-desktop` - we can't move the old ones but no need to make it stick on the new ones.

File: src/applets/night-light/IndicatorWindow.vala [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+namespace NightLight
+{
+
+[GtkTemplate (ui = "/com/solus-project/night-light/indicator_window.ui")]
```
Comment (ikeydoherty) [inline src/applets/night-light/IndicatorWindow.vala]: `/org/budgie-desktop/`

File: src/applets/night-light/indicator_window.ui [hunk 5]
```diff
@@ -0,0 +1,169 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Generated with glade 3.20.0 -->
+<interface>
+  <requires lib="gtk+" version="3.20"/>
+  <object class="GtkAdjustment" id="adjustment">
+    <property name="upper">6000</property>
```
Comment (ikeydoherty) [inline src/applets/night-light/indicator_window.ui]: Is this defined somewhere for documentation purposes.. ?

File: src/applets/night-light/indicator_window.ui [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+            <property name="can_focus">False</property>
+            <property name="margin_left">5</property>
+            <property name="margin_right">5</property>
+            <property name="row_spacing">10</property>
+            <property name="column_spacing">35</property>
+            <child>
+              <object class="GtkLabel">
+                <property name="visible">True</property>
+                <property name="can_focus">False</property>
+                <property name="tooltip_text" translatable="yes">Service to use for uploading screenshots</property>
```
Comment (ikeydoherty) [inline src/applets/night-light/indicator_window.ui]: Not really the job of a nightlight ;)

File: src/applets/night-light/meson.build [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+applet_nightlight_dir = applets_dir + '.night-light'
+
+custom_target('plugin-file-nightlight',
+    input : 'NightLight.plugin.in',
+    output : 'NightLight.plugin',
+    command : [intltool, '--desktop-style', podir, '@INPUT@', '@OUTPUT@'],
+    install : true,
+    install_dir : applet_nightlight_dir)
+
+gresource = join_paths(meson.current_source_dir(), 'com.solus-project.night-light.gresource.xml')
```
Comment (ikeydoherty) [inline src/applets/night-light/meson.build]: `/org/budgie-desktop/`

File: src/applets/night-light/indicator_window.ui (hunk 6 above)
Comment (cybre) [inline src/applets/night-light/indicator_window.ui]: Derp. Copy pasted elements xD

File: src/applets/night-light/indicator_window.ui (hunk 5 above)
Comment (cybre) [inline src/applets/night-light/indicator_window.ui]: Not defined. I tested the value and it was around the limit of my monitor so I guess it's gonna be different depending on the monitor?
Should I just remove the limit?

**Comment by ikeydoherty** [comment]
LGTM - thanks!

**Comment by fossfreedom** [comment]
Will it be possible to disable or remove the applet for distro's that link against libmutter-dev ? i.e. the plugin is only applicable for mutter v3.24 and later

**Comment by ikeydoherty** [comment]
Hm could do.

**Comment by cybre** [comment]
#895

**Pull request merged**